datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedTaskCountEstimator.java

```diff
@@ -27,6 +50,26 @@
    */
   public int getTaskCount(ClusterThroughputInfo throughputInfo, List<String> assignedPartitions,
       List<String> unassignedPartitions) {
-    throw new NotImplementedException();
+    Validate.notNull(throughputInfo, "null throughputInfo");
+    Validate.notNull(assignedPartitions, "null assignedPartitions");
+    Validate.notNull(unassignedPartitions, "null unassignedPartitions");
+
+    Map<String, PartitionThroughputInfo> throughputMap = throughputInfo.getPartitionInfoMap();
+
+    Set<String> allPartitions = new HashSet<>(assignedPartitions);
+    allPartitions.addAll(unassignedPartitions);
+
+    // total throughput in KB/sec
+    int totalThroughput = allPartitions.stream().map(p ->
+        throughputMap.getOrDefault(p, new PartitionThroughputInfo(BYTES_IN_KB_RATE_DEFAULT, MESSAGES_IN_RATE_DEFAULT, p))).
```

Comment:
Will it make more sense to create one default PartitionThroughputInfo object, rather than N objects, since we don't it for long.

Reply:
You're right. We have the name for each partition if we do it in the loop, but then again, we're not using that name.